### PR TITLE
feat!: generic Makefile/Terratest and chatops introduction

### PR DESCRIPTION
**Pull request by FoSix**
## Description

Introducing single, repository-wide `Makefile` based sub-workflows and ChatOPS reusable workflows.

## Motivation and Context

BREAKING CHANGE.

The reusable workflows shift to use a single sub-workflow that runs Terratest-based action. This sub-workflow can be used to trigger any Terratest test function by providing the test name via the `terratest_action` parameter. This is combined with the code repo based action running a unified `Makefile`. 

<details>
<summary>Example Makefile code</summary>

```Makefile
.phony: all invalidate help

all:
	@echo "Run [make help] for usage details."

invalidate:

help:
	@echo "This Makefile is run by specifying a module path as a target name." ; \
	echo "It takes one argument: ACTION. Value of this argument is specific to a particular module." ; \
	echo "It represents the name of a Terratest test function." ; \
	echo "Typically this will be: Validate, Plan, Apply, Idempotence, but it should be verified with" ; \
	echo "  module's main_test.go file." ; \
	echo ; \
	echo "Example:" ; \
	echo "  make examples/common_vmseries ACTION=Plan" ; \
	echo

%: invalidate %/main.tf
	@cd $@ && \
	echo "::group::DOWNLOADING GO DEPENDENCIES" && \
	go get -v -t -d  && \
	go mod tidy && \
	echo "::endgroup::" && \
	echo "::group::ACTION >>$(ACTION)<<" && \
	go test -run $(ACTION) -timeout 60m -count=1 && \
	echo "::endgroup::"
```
</details>

Additionally this PR introduced reusable workflow that can be used with the ChatOPS workflows in the target repository. It's logic is based on the unified, Terratest based sub-workflows described above.

## How Has This Been Tested?

Tested on Azure repo, [263-terratest-for-azure PR](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/pull/278). And on the [Azure test (internal) repository](https://github.com/PaloAltoNetworks/vm-series-gh-actions/pull/76).

## Types of changes

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/pr_ci.yml [hunk 1]
```diff
@@ -155,15 +159,22 @@ jobs:
...
+    uses: ./.github/workflows/_tf_test.yml
+    permissions:
+      contents: read
+      id-token: write
     with:
+      cloud: azure
       tf_version: ${{ inputs.tf_version }}
       paths: ${{ needs.tf_prereqs.outputs.validate_paths }}
-      max_parallel: ${{ inputs.max_parallel }}
+      terratest_action: Validate
```
Comment (sebastianczech) [inline .github/workflows/pr_ci.yml:170]: What is the reason to start `validate` from uppercase `Validate` ? In Terraform command we have `validate` - can we follow that convention ?

File: .github/workflows/_pre_commit.yml [hunk 2]
```diff
@@ -11,19 +11,17 @@ on:
...
         type: string
         required: true
       pre-commit-files:
         description: "Files for pre-commit to scan - passed dynamically via PR CI variable. "
         type: string
         default: "all-files"
         required: false
-      checkout-branch:
-        # description: 
+      branch:
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/_pre_commit.yml:22]: you mean to add `branch` as required? but it's not, you do not specify branch name, the branch from the triggering hook will be used.

File: .github/workflows/_tf_test.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ on:
         type: number
         default: 30
       branch:
-        description: Branch on which the tests shoul run
+        description: Branch on which the tests should run
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/_tf_test.yml:42]: same as above, brach is not required

File: .github/workflows/_pre_commit.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebastianczech) [reply .github/workflows/_pre_commit.yml:22]: If it's not required , can we add then `required: false` (to have the same convention as for other parameters)?

File: .github/workflows/_tf_test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebastianczech) [reply .github/workflows/_tf_test.yml:42]: Ok, but then if it's not required , can we add then `required: false` (to have the same convention as for other parameters)?

File: .github/workflows/pr_ci.yml (hunk 1 above)
Comment (sebastianczech) [reply .github/workflows/pr_ci.yml:170]: 👍

**Review by sebastianczech — APPROVED** [review]
Great work 👍

**Pull request merged**